tests/integration/golang/mlflow/experiment/set_experiment_tag_test.go

```diff
@@ -1,3 +1,208 @@
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type SetExperimentTagTestSuite struct {
+	suite.Suite
+	client   *helpers.HttpClient
+	fixtures *fixtures.ExperimentFixtures
+}
+
+func TestSetExperimentTagTestSuite(t *testing.T) {
+	suite.Run(t, new(SetExperimentTagTestSuite))
+}
+
+func (s *SetExperimentTagTestSuite) SetupTest() {
+	s.client = helpers.NewHttpClient(os.Getenv("SERVICE_BASE_URL"))
+	fixtures, err := fixtures.NewExperimentFixtures(os.Getenv("DATABASE_DSN"))
+	assert.Nil(s.T(), err)
+	s.fixtures = fixtures
+}
+
+func SetExperimentTag(s *SetExperimentTagTestSuite, experiment *models.Experiment, key, value string) {
+	req := request.SetExperimentTagRequest{
+		ID:    fmt.Sprintf("%d", *experiment.ID),
+		Key:   key,
+		Value: value,
+	}
+	resp := fiber.Map{}
+	err := s.client.DoPostRequest(
+		fmt.Sprintf(
+			"%s%s", mlflow.ExperimentsRoutePrefix, mlflow.ExperimentsSetExperimentTag,
+		),
+		req,
+		&resp,
+	)
+	assert.Nil(s.T(), err)
+}
+
+func (s *SetExperimentTagTestSuite) Test_Ok() {
+	// 1. prepare database with test data.
+	experiment, err := s.fixtures.CreateTestExperiment(context.Background(), &models.Experiment{
+		Name: "Test Experiment",
+		CreationTime: sql.NullInt64{
+			Int64: time.Now().UTC().UnixMilli(),
+			Valid: true,
+		},
+		LastUpdateTime: sql.NullInt64{
+			Int64: time.Now().UTC().UnixMilli(),
+			Valid: true,
+		},
+		LifecycleStage:   models.LifecycleStageActive,
+		ArtifactLocation: "/artifact/location",
+	})
+	assert.Nil(s.T(), err)
+	experiment1, err := s.fixtures.CreateTestExperiment(context.Background(), &models.Experiment{
+		Name: "Test Experiment2",
+		CreationTime: sql.NullInt64{
+			Int64: time.Now().UTC().UnixMilli(),
+			Valid: true,
+		},
+		LastUpdateTime: sql.NullInt64{
+			Int64: time.Now().UTC().UnixMilli(),
+			Valid: true,
+		},
+		LifecycleStage:   models.LifecycleStageActive,
+		ArtifactLocation: "/artifact/location",
+	})
+	assert.Nil(s.T(), err)
+	defer func() {
+		assert.Nil(s.T(), s.fixtures.UnloadFixtures())
+	}()
+
+	// Set tag on experiment
+	SetExperimentTag(s, experiment, "KeyTag1", "ValueTag1")
+	exp, err := s.fixtures.GetExperimentByID(context.Background(), *experiment.ID)
+	assert.Nil(s.T(), err)
+
+	found := false
+	for _, tag := range exp.Tags {
+		if tag.Key == "KeyTag1" && tag.Value == "ValueTag1" {
+			found = true
+			break
+		}
+	}
+	assert.True(s.T(), found, "Expected 'experiment.tags' to contain 'KeyTag1' with value 'ValueTag1'")
+
+	// Update tag on experiment
+	SetExperimentTag(s, experiment, "KeyTag1", "ValueTag2")
+	exp, err = s.fixtures.GetExperimentByID(context.Background(), *experiment.ID)
+	assert.Nil(s.T(), err)
+
+	found = false
+	for _, tag := range exp.Tags {
+		if tag.Key == "KeyTag1" && tag.Value == "ValueTag2" {
+			found = true
+			break
+		}
+	}
```

Comment:
Let put such functions to `helper` file near the test. In that case your test will be much cleaner and understandable.